### ISSUE TITLE
Report expected location of registry backend in UI

### ISSUE THREAD

**Issue by pwright** [Apicurio/apicurio-registry#5388]
### Feature or Problem Description

With v3, there is the possibility of users not configuring the UI correctly and getting message:

Connection to the Registry server failed (could not reach the server). Please check your connection and try again, or report this error to an admin.

When clicking help, this modal is displayed:
![image](https://github.com/user-attachments/assets/312beb62-e001-4318-9943-f1ad68750146)

### Proposed Solution

On one of the screens, show the user the current configuration, ie where the UI expects to connect to, eg localhost:8080

```
$ more /opt/app-root/src/config.js

const ApicurioRegistryConfig = {
    "artifacts": {
        "url": "http://localhost:8080/apis/registry/v3"
    },
    "ui": {},
    "auth": {},
    "features": {}
};
```

**Comment by simonepm** [comment]
Hello, even if the endpoint is reachable (see curl) and the configured endpoint matches, I still receive: "Connection to the Registry server failed (could not reach the server). Please check your connection and try again, or report this error to an admin."

![image](https://github.com/user-attachments/assets/e524526f-2a48-4b42-afa6-6e9a90e0da55)

What I am missing?
Thanks!

**Comment by devops-42** [comment]
@simonepm Have same issue with Kubernetes deployments. When looking into the browser's inspector I saw, that the UI pod requests the browser to directly call the configured Kubernetes internal service URL, which fails of course. Maybe there's a config parameter where URL rewriting / proxying can be configured, but didn't found it in the documentation yet.

@EricWittmann Do you have any additional links or information? Thanks a lot in advance!

**Comment by EricWittmann** [comment]
> even if the endpoint is reachable (see curl) and the configured endpoint matches, I still receive:
> 
> "Connection to the Registry server failed (could not reach the server). Please check your connection and try again, or report this error to an admin."
>
> What am I missing?

Ack, I didn't see this until today!  Sorry about that.

Typically if you can reach the API via curl but not in the UI then you have a CORS issue.  You'll need to configure `QUARKUS_HTTP_CORS_ORIGINS` in the app deployment/pod to allow connections from the UI.  You can either set the value to `*` or to the `scheme://host` of your UI.

I would recommend starting with `*` to see if that works (very permissive).  If it does, then you could try setting it to the specific external URL of your UI (e.g. the ingress host) and see if you can get it working that way.

**Comment by EricWittmann** [comment]
> [@simonepm](https://github.com/simonepm) Have same issue with Kubernetes deployments. When looking into the browser's inspector I saw, that the UI pod requests the browser to directly call the configured Kubernetes internal service URL, which fails of course. Maybe there's a config parameter where URL rewriting / proxying can be configured, but didn't found it in the documentation yet.
> 
> [@EricWittmann](https://github.com/EricWittmann) Do you have any additional links or information? Thanks a lot in advance!
> 
> Used versions:
> 
>     * `apicurio-registry: 3.0.6`
> 
>     * `apicurio-registry-ui: 3.0.6`

In this case you need to configure the UI to know where to find the API.  This is done by configuring the UI pod/deployment using this env var:

`REGISTRY_API_URL`

Set that to the external URL of your registry-api service. 

Reference:  https://github.com/Apicurio/apicurio-registry/blob/main/ui/.docker-scripts/create-config.cjs#L9

**Comment by devops-42** [comment]
> > [@simonepm](https://github.com/simonepm) Have same issue with Kubernetes deployments. When looking into the browser's inspector I saw, that the UI pod requests the browser to directly call the configured Kubernetes internal service URL, which fails of course. Maybe there's a config parameter where URL rewriting / proxying can be configured, but didn't found it in the documentation yet.
> > [@EricWittmann](https://github.com/EricWittmann) Do you have any additional links or information? Thanks a lot in advance!
> > Used versions:
> > ```
> > * `apicurio-registry: 3.0.6`
> > 
> > * `apicurio-registry-ui: 3.0.6`
> > ```
> 
> In this case you need to configure the UI to know where to find the API. This is done by configuring the UI pod/deployment using this env var:
> 
> `REGISTRY_API_URL`
> 
> Set that to the external URL of your registry-api service.
> 
> Reference: https://github.com/Apicurio/apicurio-registry/blob/main/ui/.docker-scripts/create-config.cjs#L9
> 
> If you look at the `config.js` file that is loaded by the UI, you can see where it's getting that from.

@EricWittmann Thanks for your response!

What I did: Have deployed the apicurio-registry in Kubernetes and created a service for accessing the API. Then I deployed the UI pod and configured the `REGISTRY_API_URL` pointing to the service name and port (`apicurio-registry:8080`). For accessing the UI I created an ingress resource pointing to the service of the UI.

What I've seen so far: Calling the URL of the UI ingress brings the same output as @simonepm has shown. When I open the network inspector console of the browser I saw, that the browser tries the URL `apicurio-registry:8080` which is impossible from outside the Kubernetes cluster.
I created a second setup where I run the apicurio-registry locally on port 8080 and configured the `REGISTRY_API_URL` with `localhost:8080`. This worked so far, but this might not be the target deployment setup.

Maybe I miss something here and would appreciate any help!

Thanks!

**Comment by EricWittmann** [comment]
> I've seen you mentioned the external URL, so I'd need here to expose the Apicurio API Url to the public even if it's only needed from inside the cluster?

Yes this is it exactly. You need an ingress for the API. Because the user interface makes API calls directly from the browser. The user interface does not have its own backend component. So yes, the API has to be publicly accessible so that the UI can access it.

This means you have to configure the UI with that external URL. You also need to configure CORS allowed origins to include the external URL of the UI. Or of course you could just set it to *.

**Comment by cgpoh** [comment]
Hi @EricWittmann , my api ingress is using keycloak, what are the configurations to make the ui work? I created 2 ingresses, 1 for the ui and 1 for the api and both ingresses are using keycloak. Thanks!
